packages/Pagination/README.md

```diff
@@ -0,0 +1,132 @@
+| `pageLimit`           | `number` | integer representing number of records to display per page         | `30`                |
+| `pageNeighbours`      | `number` | integer representing number of links on either side of active link | `2`                 |
+| `previousLabel`       | `string` | label for pagination item linking to the previous page             | `Previous`          |
+| `startLabel`          | `string` | label for pagination item linking to the first page                | `Start`             |
+| `totalRecords`        | `number` | integer representing number of all records                         | `0`                 |
+
+#### **Paginator Component**
+
+```typescript
+import { Paginator } from '@onaio/pagination';
+
+<Paginator />;
+```
+
+##### _props for Paginator_
+
+Further customizable options for Paginator include:
+
+| PropName       | type                          | description                                                                          |
+| -------------- | ----------------------------- | ------------------------------------------------------------------------------------ |
+| `onPageChange` | `(e: paginationData) => void` | function that is called when a link is clicked, see below for info on paginationData |
+
+**PaginationData**:
+
+This documents the structure of the data object that is given as the first and only argument given to the onPageChange callback handler
+
+| Property       | type     | description                                                |
+| -------------- | -------- | ---------------------------------------------------------- |
+| `currentPage`  | `number` | Page number of the currently active link                   |
+| `pageLimit`    | `number` | integer representing number of records to display per page |
+| `totalPages`   | `number` | the total number of pages                                  |
+| `totalRecords` | `number` | integer representing number of all records                 |
+
+#### Paginator Code example
+
+```typescript
+import { Paginator, PaginatorProps } from '@onaio/pagination';
+
+const props: PaginatorProps = {
+  ariaLabel: 'pagination sample',
+  onPageChange: data => alert(`clicked page no ${data.currentPage}`),
+  pageLimit: 35,
+  pageNeighbours: 3,
+  totalRecords: 589
+};
+
+<Paginator {...props} />;
+```
+
+#### **RoutedPaginator Component**
+
+```typescript
+import { RoutedPaginator } from '@onaio/pagination';
+
+<RoutedPaginator />;
+```
+
+##### _props for RoutedPaginator_
+
+Further customizable options for RoutedPaginator include:
+
+| PropName | type     | description                                                                             | default  |
+| -------- | -------- | --------------------------------------------------------------------------------------- | -------- |
+| `urlKey` | `string` | denotes the key in the the registered routes path whose value will be used as the page, | `"page"` |
```

Comment:
What this props does is a little difficult to understand for me.

Reply:
made some changes, Please see if it makes more sense now

Reply:
super clear now :smiley: